torch/_dynamo/source.py

```diff
@@ -466,6 +466,15 @@ def guard_source(self):
         return GuardSource.GLOBAL
 
 
+@dataclasses.dataclass(frozen=True)
+class TorchFunctionStateSource(Source):
+    def name(self):
+        return ""
+
+    def guard_source(self) -> GuardSource:
+        return GuardSource.GLOBAL
```

Comment:
We have a pile of these and I hate them all with their `return ""` - add a GlobalStateSource that does the same thing and refactor GradModeSource, DeterministicAlgorithmsSource, DefaultDeviceSource, this, and whatever else into it.

Also, invoking name here should be illegal, not empty, right?

Reply:
Yeah I can refactor these, I considered it but was lazy haha